flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -205,12 +207,34 @@ private Object invokeRpc(Method method, Object[] args) throws Exception {
 			result = null;
 		} else if (Objects.equals(returnType, CompletableFuture.class)) {
 			// execute an asynchronous call
-			result = ask(rpcInvocation, futureTimeout);
+			CompletableFuture resultFuture = ask(rpcInvocation, futureTimeout);
+
+			result = resultFuture.thenApply((Object o) -> {
+				if (o instanceof SerializedValue) {
+					try {
+						return  ((SerializedValue) o).deserializeValue(getClass().getClassLoader());
+					} catch (IOException | ClassNotFoundException e) {
+						throw new CompletionException(e);
```

Comment:
The same applies here. It would be good `CompletionExcpetion(new RpcException("Could not deserialize...", e));`